x-pack/functionbeat/magefile.go

```diff
@@ -99,5 +135,61 @@ func GoTestIntegration(ctx context.Context) error {
 
 // Config generates both the short and reference configs.
 func Config() error {
-	return devtools.Config(devtools.ShortConfigType|devtools.ReferenceConfigType, functionbeat.XPackConfigFileParams(), ".")
+	for _, provider := range selectedProviders {
+		devtools.BeatIndexPrefix += "-" + provider
+		err := devtools.Config(devtools.ShortConfigType|devtools.ReferenceConfigType, functionbeat.XPackConfigFileParams(provider), provider)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Fields generates a fields.yml for the Beat.
+func Fields() error {
+	for _, provider := range selectedProviders {
+		output := filepath.Join(devtools.CWD(), provider, "fields.yml")
+		err := devtools.GenerateFieldsYAMLTo(output)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func includeFields() error {
+	fnBeatDir := devtools.CWD()
+	for _, provider := range selectedProviders {
+		err := os.Chdir(filepath.Join(fnBeatDir, provider))
```

Comment:
Changing directories should hopefully not be necessary since the underlying `asset.go` script allows for the location of `fields.yml` to be specified. Running a command like this should work 🤞  

`devtools.GenerateFieldsGoWithName(devtools.BeatName+"-"+provider, "provider/aws/fields.yml", "provider/aws/include/fields.go")`